data/helpers.d/setting

```diff
@@ -179,27 +189,72 @@ ynh_webpath_register () {
 # Requires YunoHost version 3.7.0 or higher.
 ynh_permission_create() {
     # Declare an array to define the options of this helper.
-    local legacy_args=pua
-    local -A args_array=( [p]=permission= [u]=url= [a]=allowed= )
+    local legacy_args=puAhaltP
+    declare -A args_array=( [p]=permission= [u]=url= [A]=additional_urls= [h]=auth_header= [a]=allowed= [l]=label= [t]=show_tile= [P]=protected= )
     local permission
     local url
+    local additional_urls
+    local auth_header
     local allowed
+    local label
+    local show_tile
+    local protected
     ynh_handle_getopts_args "$@"
     url=${url:-}
+    additional_urls=${additional_urls:-}
+    auth_header=${auth_header:-}
     allowed=${allowed:-}
+    label=${label:-}
+    show_tile=${show_tile:-}
+    protected=${protected:-}
 
     if [[ -n $url ]]
     then
-        url="'$url'"
-    else
-        url="None"
+        url=",url='$url'"
+    fi
+
+    if [[ -n $additional_urls ]]
+    then
+        additional_urls=",additional_urls=['${additional_urls//';'/"','"}']"
```

Comment:
The example use `--additional_urls 'domain.tld/otherurl /superadmin'`, only one value.
The variable is set with `additional_urls=${additional_urls:-}` which will lose the second path...
And here `;` are replaced by ','

Are the separate urls supposed to be splitted by `;` ? No handling of multiple urls anywhere.

And I have to admit that the comment, just before that line is hard to read...

If the purpose (I guess...?) was to use the multiple value feature of getopts, as the **comment** explains https://github.com/YunoHost/yunohost/blob/stretch-unstable/data/helpers.d/getopts#L40
First do not put quotes on the options during the call for the helper, otherwise it become only one value.
And then, replace `;` not `';'`
And finally... are you sure the $additional_urls is interpreted between simple quotes ?

Reply:
Well,
Yes the first comment (`--additional_urls 'domain.tld/otherurl /superadmin'`) is false, I did probably some change and forgot to change also here...

But if you use correctly this helper it work. I've tested in the test app. cf https://github.com/YunoHost/test_apps/blob/permission_protection/permissions_app_ynh/scripts/install#L29

Yes I admin that we need some comment. Maybe a global comment could do this because it's everywhere the same (tricky) thing.

So do you prefer this ?
```suggestion
        additional_urls=",additional_urls=['${additional_urls//;/"','"}']"
```

Reply:
I "do prefer" something that actually works...
So my question, does it work with that code !?

Damn is that really so difficult to explain your code !!!???

Is the expected result is that ?
`,additional_urls=['url1','url2','url3']`
If so, I do prefer a code easy to read (as usual...) like
`additional_urls=",additional_urls=['${additional_urls//;/\',\'}']"`
Except the first and final quote, the adding of quote around the comma is clear.
Which does not spare a comment to explain.

Reply:
> Yes I admit that we need some comment. Maybe a global comment could do this because it's everywhere the same (tricky) thing.

Are comments especially expensive in your IDE ?
Comment everywhere anyone could wonder about your code.